### DISCUSSION TITLE
How to select the use of g2o and gtsam?

### DISCUSSION THREAD

**Discussion by zhanghua7099** [stella-cv/stella_vslam#542] [Q&A] [answered]
Hi! Thanks for your great work.

I found that this project seems to support g2o and gtsam at the same time.

https://github.com/stella-cv/stella_vslam/blob/eb1c04488a023b1f7a6549f68ada621df123ac16/src/stella_vslam/CMakeLists.txt#L13C2-L24

https://github.com/stella-cv/stella_vslam/blob/eb1c04488a023b1f7a6549f68ada621df123ac16/src/stella_vslam/CMakeLists.txt#L26C4-L36

Could you please tell me how to specify gtsam? Thank you very much!

**Comment by Skwangles** [comment, marked as answer]
In the config you would put 'gtsam' instead of g2o

e.g. https://stella-cv.readthedocs.io/en/latest/parameters.html#tracking

Have a look through this example and just see where you can ctrl+f and replace g2o with gtsam https://github.com/stella-cv/stella_vslam/blob/main/example/euroc/EuRoC_mono.yaml

**Reply by zhanghua7099** [reply]
Thanks for the info. The documentation has been updated. It's very clear now. 

By the way, I have one more question. Will stella_vslam perform differently using g2o and gtsam as the backend?

**Reply by Skwangles** [reply]
I am not too sure on the performance differences, my use case doesn't care too much so I just use g2o. gtsam and g2o will have different strengths, so performance will be use case specific.

If you wanted, you could try benchmarking them on some of the example datasets

**Reply by zhanghua7099** [reply]
Thank you for your reply! I will try to evaluate the difference using a public dataset.

**Comment by zhanghua7099** [comment]
Since the document has been updated, this discussion will be closed. Thanks for the valuable discussion!

**Reply by ZhiweiHuang5520** [reply]
Hi, zhanghua, did you benchmark the two different backends and feel comfortable to share the results?
